xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/IQAdminHandler.java

```diff
@@ -98,6 +98,8 @@ public void handleIQ(IQ packet, MUCRole role) throws ForbiddenException, Conflic
         
         if (!itemsList.isEmpty()) {
             handleItemsElement(role, itemsList, reply);
+        } else if (!itemsList.isEmpty() && !itemsList.toString().contains(packet.getFrom().toBareJID())) {
```

Comment:
This will never be invoked. When the condition that is added here evaluates to `true`, then the condition in the `if` statement just above the one that you added will also be true. That will prevent the `else` statement from being invoked.

Reply:
The way in which this code checks if a particular bare JID is in a list of elements is very unsafe. Doing a `toString` on the collection might, depending on the implementation of the collection, not even give a result that reflects the String-based equivalent of each of the elements. This should be replaced by a more explicit check that iterates over each element in the collection, somehow.

Reply:
> @lakadbagha: Have you seen comments?

Hi am very sorry for busy. I get new mess in openfire suppose if i joined any conference and my network start dancing(fluctuate) then ID get error. And then ID not joining and getting room info ever again. I asked in community and nobody replied me there.

Reply:
@guusdk Can you please review and update the code.

if (!itemsList.isEmpty() && packet.getFrom() == null) {
          // Case when openfire admin portal is used to modify role.
            handleItemsElement(role, itemsList, reply);
        } else if (!itemsList.isEmpty() && !itemsList.toString().contains(packet.getFrom().toBareJID())) {
         // Case when client try to modify the role.
            handleItemsElement(role, itemsList, reply);
        } else {
            // An unknown and possibly incorrect element was included in the query
            // element so answer a BAD_REQUEST error
            reply.setChildElement(packet.getChildElement().createCopy());
            reply.setError(PacketError.Condition.bad_request);
        }